sys/kern/uipc_ktls.c

```diff
@@ -353,7 +354,8 @@ ktls_buffer_import(void *arg, void **store, int count, int domain, int flags)
 		    VM_MEMATTR_DEFAULT);
 		if (m == NULL)
 			break;
-		store[i] = (void *)PHYS_TO_DMAP(VM_PAGE_TO_PHYS(m));
+		item = (void *)PHYS_TO_DMAP(VM_PAGE_TO_PHYS(m));
+		store[i] = cheri_kern_setbounds(item, ktls_maxlen);
```

Comment:
I would suggest avoiding the temporary and instead just adding the line

```
store[i] = cheri_kern_setbounds(store[i], ktls_maxlen);
```

That will be less prone to conflicts for future merges.

Reply:
Fixed.